libsrc/Helper.cpp

```diff
@@ -103,9 +103,32 @@ namespace dcmqi {
 
 
   string Helper::floatToStrScientific(float f) {
+    /*
+    Alternatife with scientific to get to 16:
+     - mantissa sign (1)
+     - mantissa leading number (1)
+     - dot (1)
+     - mantissa after the dot (8)
+     - E (1)
+     - exponent sign (1)
+     - exponent (2 OR 3 (Win))
+
+    ostringstream sstream;
+    sstream << setprecision(8) << scientific << f;
+    cout << setprecision(8) << scientific << f << endl;
+    */
     ostringstream sstream;
-    sstream << scientific << f;
-    return sstream.str();
+    sstream << fixed << setprecision(15) << f;
+    string formatted_f = sstream.str();
+    size_t dot_position = formatted_f.find('.');
+    if(formatted_f.length()>16){
+      if(dot_position == string::npos || dot_position > 15){
+        cerr << "ERROR: Failed to convert " << f << " to DS VR!" << endl;
+      } else {
+        formatted_f = formatted_f.substr(0,16);
+      }
+    }
```

Comment:
Should not be needed, if everything works according to standard. But it doesn't hurt, if you have no very high performance requirements.